### DISCUSSION TITLE
v2.0.0-beta

### DISCUSSION THREAD

**Discussion by hrj** [librecaptcha/lc-core#154] [General]
# v2.0.0 beta

> This release contains some breaking changes, hence bumping the version number to 2!

## 💔 Breaking changes
* API is now served on `/v2` path. A new parameter (`size`) is required for `/v2/captcha` endpoint.
* Config option `throttle` has been renamed to `bufferCount` for clarity
* Config option `maxAttempts` has been replaced by `maxAttemptsRatio` to simplify management of server. The parameter is now expressed as a fraction of the `bufferCount`. This allows you to change `bufferCount` in response to server load, without worrying about changing `maxAttempts` to match it. 
* New config option `allowedSizes` lets you define the size of the captcha image, per captcha provider.

## Fixes
See https://github.com/librecaptcha/lc-core/milestone/5?closed=1

## Changes Behind the scenes
* Updated to Scala  3
* Updated H2 to 2.0.206

## Detailed commit log
* Update json4s-jackson_2.13 to 3.6.12 by @scala-steward in https://github.com/librecaptcha/lc-core/pull/106
* Update sbt to 1.5.7 by @scala-steward in https://github.com/librecaptcha/lc-core/pull/107
* Update scalafmt-core to 3.0.8 by @scala-steward in https://github.com/librecaptcha/lc-core/pull/108
* Update scalafmt-core to 3.2.1 by @scala-steward in https://github.com/librecaptcha/lc-core/pull/110
* Update sbt to 1.5.8 by @scala-steward in https://github.com/librecaptcha/lc-core/pull/111
* update H2 lib to 2.0.204 by @hrj in https://github.com/librecaptcha/lc-core/pull/112
* Update scalafmt-core to 3.2.2 by @scala-steward in https://github.com/librecaptcha/lc-core/pull/113
* Update sbt-scalafmt to 2.4.6 by @scala-steward in https://github.com/librecaptcha/lc-core/pull/114
* Update scalafmt-core to 3.3.0 by @scala-steward in https://github.com/librecaptcha/lc-core/pull/115
* Update sbt to 1.6.0 by @scala-steward in https://github.com/librecaptcha/lc-core/pull/117
* Max attempts config by @rr83019 in https://github.com/librecaptcha/lc-core/pull/116
* Update json4s-jackson_2.13 to 4.0.3 by @scala-steward in https://github.com/librecaptcha/lc-core/pull/109
* Update sbt to 1.6.1 by @scala-steward in https://github.com/librecaptcha/lc-core/pull/118
* Update scalafmt-core to 3.3.1 by @scala-steward in https://github.com/librecaptcha/lc-core/pull/119
* update to scala 3 by @hrj in https://github.com/librecaptcha/lc-core/pull/120
* update jar path in docker file by @hrj in https://github.com/librecaptcha/lc-core/pull/121
* Update h2 lib to 2.0.206 by @hrj in https://github.com/librecaptcha/lc-core/pull/123
* Make config parameters optional by @rr83019 in https://github.com/librecaptcha/lc-core/pull/122
* Update scrimage-core, scrimage-filters to 4.0.25 by @scala-steward in https://github.com/librecaptcha/lc-core/pull/124
* Update sbt-scalafix to 0.9.34 by @scala-steward in https://github.com/librecaptcha/lc-core/pull/125
* Update scalafmt-core to 3.3.2 by @scala-steward in https://github.com/librecaptcha/lc-core/pull/127
* Update scalafmt-core to 3.3.3 by @scala-steward in https://github.com/librecaptcha/lc-core/pull/128
* Update json4s-jackson to 4.0.4 by @scala-steward in https://github.com/librecaptcha/lc-core/pull/129
* Update scrimage-core, scrimage-filters to 4.0.26 by @scala-steward in https://github.com/librecaptcha/lc-core/pull/130
* Update scalafmt-core to 3.4.0 by @scala-steward in https://github.com/librecaptcha/lc-core/pull/131
* Update sbt to 1.6.2 by @scala-steward in https://github.com/librecaptcha/lc-core/pull/132
* Update scala3-library to 3.1.1 by @scala-steward in https://github.com/librecaptcha/lc-core/pull/133
* Update scalafmt-core to 3.4.1 by @scala-steward in https://github.com/librecaptcha/lc-core/pull/134
* Update scalafmt-core to 3.4.2 by @scala-steward in https://github.com/librecaptcha/lc-core/pull/135
* Update scalafmt-core to 3.4.3 by @scala-steward in https://github.com/librecaptcha/lc-core/pull/136
* Update sbt-assembly to 1.1.1 by @scala-steward in https://github.com/librecaptcha/lc-core/pull/137
* Update sbt-assembly to 1.2.0 by @scala-steward in https://github.com/librecaptcha/lc-core/pull/138
* Update scrimage-core, scrimage-filters to 4.0.27 by @scala-steward in https://github.com/librecaptcha/lc-core/pull/139
* Update scrimage-core, scrimage-filters to 4.0.28 by @scala-steward in https://github.com/librecaptcha/lc-core/pull/141
* Update scrimage-core, scrimage-filters to 4.0.29 by @scala-steward in https://github.com/librecaptcha/lc-core/pull/142
* Update scrimage-core, scrimage-filters to 4.0.30 by @scala-steward in https://github.com/librecaptcha/lc-core/pull/143
* Simplification and minor refactoring by @hrj in https://github.com/librecaptcha/lc-core/pull/144
* Update scrimage-core, scrimage-filters to 4.0.31 by @scala-steward in https://github.com/librecaptcha/lc-core/pull/147
* Fix for #145: frequent repetition of CAPTCHA challenge by @hrj in https://github.com/librecaptcha/lc-core/pull/148
* Bump version numbers, fixes #149 by @Korkman in https://github.com/librecaptcha/lc-core/pull/150
* Point docker-compose.yml to "latest", add "build" by @Korkman in https://github.com/librecaptcha/lc-core/pull/151
* Update scalafmt-core to 3.5.0 by @scala-steward in https://github.com/librecaptcha/lc-core/pull/152
* Respect size param by @hrj in https://github.com/librecaptcha/lc-core/pull/153

## New Contributors
* @scala-steward made their first contribution in https://github.com/librecaptcha/lc-core/pull/106
* @Korkman made their first contribution in https://github.com/librecaptcha/lc-core/pull/150

**Full Changelog**: https://github.com/librecaptcha/lc-core/compare/v1.1.0...v2.0.0-beta

<hr /><em>This discussion was created from the release <a href='https://github.com/librecaptcha/lc-core/releases/tag/v2.0.0-beta'>v2.0.0-beta</a>.</em>

**Comment by hrj** [comment]
CC @Korkman @kringo @JacobPozaic 

Please try this beta release and let us know your feedback if any! Will make a final release after getting some ✅

**Reply by blackmagic0** [reply]
Seems to be working good this far.  I have posted a PR to update the README.md file.

**Reply by kringo** [reply]
Just tested size and it works like a charm. 

Added couple of sizes, requested with those size and it return captcha images as expected.

Note: I think previously in v1 the size param was an object with width and height props and now it's just a string like 250x75 which is simpler and easier to use IMHO

Thanks & nicely done @hrj

**Reply by hrj** [reply]
Thanks for the feedback! Have published the final release 🎉
